lib/configure.js

```diff
@@ -310,6 +311,50 @@ function configure (gyp, argv, callback) {
 
 }
 
+/**
+ * Returns the first file or directory from an array of candidates that is 
+ * readable by the current user, or undefined if none of the candidates are
+ * readable. 
+ */
+function findAccessibleSync (logprefix, dir, candidates) {
+  var found = undefined
+  var stats, candidate
+  for (var next = 0; next < candidates.length; next++) {
+     candidate = path.resolve(dir, candidates[next])
+     try {
+       stats = fs.statSync(candidate)
+     } catch (exception) {
+       // this candidate was not found or not readable, do nothing
+       log.silly(logprefix, '%s does not exist', candidate)
+       continue
+     }
+     if (stats.uid === process.getuid()) {
+       if (stats.mode & 0400) {
+         log.silly(logprefix, '%s has owner read permission', candidate)
+         found = candidate
+         break
+       }
+     } else if (process.getgroups().some(function (group) {
+         return stats.gid === group
+       }, false)) {
+       if (stats.mode & 0040) {
+         log.silly(logprefix, '%s has group read permission', candidate)
+         found = candidate
+         break
+       }
+     } else {
+       if (stats.mode & 0004) {
+         log.silly(logprefix, '%s has others read permission', candidate)
+         found = candidate
+         break
+       }
+     }
+     log.silly(logprefix, '%s exists but has no read permissions', candidate)
+  }
+
+  return found
+}
```

Comment:
Question: if the sole purpose is to find out if the file can be read from, why don't you try to open it in read-only mode?  It's shorter and less error-prone, IMO.
```js
for (var i = 0; i < candidates.length; i += 1) {
  var filename = path.resolve(candidates[i])
  try {
    var fd = fs.openSync(filename, 'r')
  } catch (e) {
    continue
  }
  fs.closeSync(fd)
  return filename
}
```